src/Components/Server/src/Circuits/CircuitMetrics.cs

```diff
@@ -0,0 +1,110 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Components.Server.Circuits;
+
+internal sealed class CircuitMetrics : IDisposable
+{
+    public const string MeterName = "Microsoft.AspNetCore.Components.Server.Circuits";
+
+    private readonly Meter _meter;
+    private readonly Counter<long> _circuitTotalCounter;
+    private readonly UpDownCounter<long> _circuitActiveCounter;
+    private readonly UpDownCounter<long> _circuitConnectedCounter;
+    private readonly Histogram<double> _circuitDuration;
+
+    public CircuitMetrics(IMeterFactory meterFactory)
+    {
+        Debug.Assert(meterFactory != null);
+
+        _meter = meterFactory.Create(MeterName);
+
+        _circuitTotalCounter = _meter.CreateCounter<long>(
+            "aspnetcore.components.circuits.count",
+            unit: "{circuits}",
+            description: "Number of active circuits.");
+
+        _circuitActiveCounter = _meter.CreateUpDownCounter<long>(
+            "aspnetcore.components.circuits.active_circuits",
+            unit: "{circuits}",
+            description: "Number of active circuits.");
+
+        _circuitConnectedCounter = _meter.CreateUpDownCounter<long>(
+            "aspnetcore.components.circuits.connected_circuits",
+            unit: "{circuits}",
+            description: "Number of disconnected circuits.");
+
+        _circuitDuration = _meter.CreateHistogram<double>(
+            "aspnetcore.components.circuits.duration",
+            unit: "s",
+            description: "Duration of circuit.",
+            advice: new InstrumentAdvice<double> { HistogramBucketBoundaries = MetricsConstants.VeryLongSecondsBucketBoundaries });
+    }
+
+    public void OnCircuitOpened()
+    {
+        var tags = new TagList();
```

Comment:
Can this be a private static field, since we never change it from the default value.

Reply:
It doesn't need tags at all at the moment.